logic/service/raw-data-service.js

```diff
@@ -90,7 +113,48 @@ exports.RawDataService = DataService.specialize(/** @lends RawDataService.protot
      */
     fetchObjectProperty: {
         value: function (object, propertyName) {
-            return this.nullPromise;
+            var blueprint = this._bluePrintForObject(object),
+                propertyBlueprint = blueprint && blueprint.propertyBlueprintForName(propertyName),
+                propertyDescriptor = propertyBlueprint && propertyBlueprint.valueDescriptor;
+            return propertyDescriptor ? this._fetchObjectPropertyWithDescriptor(object, propertyName, propertyDescriptor) :
+                                        this.nullPromise;
+        }
+    },
+
+    _fetchObjectPropertyWithDescriptor: {
+        value: function (object, propertyName, propertyDescriptor) {
+            var service = this.rootService;
+            return this._blueprintTypeForValueDescriptor(propertyDescriptor).then(function (type) {
+                var selector = DataSelector.withTypeAndCriteria(type, {
+                    source: object,
+                    relationshipKey: propertyName
+                });
+                return service.fetchData(selector);
+            });
+        }
+    },
+
+    _bluePrintForObject: {
+        value: function (object) {
+            var typeName = object.constructor.TYPE.typeName,
```

Comment:
OK - I think I'm still one or two iterations away from the goal but it is starting to come into picture.  A business object will have an object descriptor and a set of mappings.  There should be one set of mappings for raw to cooked and another for cooked to raw.

A mapping has a propertyName, an expression and a value converter?  When the mappings are created the expression is compiled and the raw/cooked value is passed in as the scope?